timesketch/api/v1/resources/sigma.py

```diff
@@ -87,15 +125,274 @@ def get(self, rule_uuid):
+        is required to parse the rule.
+
+        Args:
+            rule_uuid: uuid of the rule
+
+        Returns:
+            Sigma rule object and HTTP status code indicating
+            whether operation was sucessful.
+        """
+        form = request.json
+        if not form:
+            form = request.data
+
+        rule_yaml = form.get("rule_yaml")
+
+        if not rule_yaml:
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                "No rule_yaml supplied in the POST request to the API.",
+            )
+
+        if not isinstance(rule_yaml, str):
+            abort(
+                HTTP_STATUS_CODE_FORBIDDEN, "rule_yaml needs to be a string."
+            )
+
+        parsed_rule = ts_sigma_lib.parse_sigma_rule_by_text(rule_yaml)
+
+        if not rule_uuid:
+            rule_uuid = parsed_rule.get("id")
+
+        # Query rules to see if it already exist and exit if found
+        sigma_rule_from_db = SigmaRule.query.filter_by(
+            rule_uuid=rule_uuid
+        ).first()
+        if sigma_rule_from_db:
+            if sigma_rule_from_db.rule_uuid == parsed_rule.get("rule_uuid"):
+                logger.debug("Rule was already found in teh database")
+                # it will overwritten with the provided content
+
+        title = parsed_rule.get("title")
+        if not title:
+            logger.error("Sigma rule without title was given")
+            return abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Missing value: 'title' from the request.",
+            )
+
+        description = parsed_rule.get("description")
+        if not description:
+            return abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Missing value: 'description' from the request.",
+            )
+
+        sigma_rule = SigmaRule.get_or_create(
+            rule_uuid=rule_uuid,
+            rule_yaml=form.get("rule_yaml"),
+            description=description,
+            title=title,
+            user=current_user,
+        )
+
+        if not sigma_rule:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No sigma rule object created")
+
+        sigma_rule.query_string = parsed_rule.get("es_query")
+        sigma_rule.rule_uuid = parsed_rule.get("id")
+        sigma_rule.set_status(parsed_rule.get("status", "experimental"))
+
+        try:
+            db_session.add(sigma_rule)
+            db_session.commit()
+        except IntegrityError as e:
+            logger.error("Unable to add Sigma rule to DB, with error: %s", e)
+            abort(
+                HTTP_STATUS_CODE_CONFLICT,
+                "Problem adding Sigma rule {0!s}".format(e),
+            )
+
+        return self.to_json(sigma_rule, status_code=HTTP_STATUS_CODE_CREATED)
+
+
+class SigmaRuleByTextResource(resources.ResourceMixin, Resource):
+    """Resource to get a Sigma rule by text."""
+
     @login_required
     def post(self):
```

Comment:
What does this function do? It returns a parsed Sigma Rule based on the body of the POST request? Please make the docstring explicit.